atom_gems/AtomTutorials/Assets/VegetationBending/VegetationBending_ForwardPass_EDS.shader

```diff
@@ -0,0 +1,49 @@
+{
+    "Source" : "./VegetationBending_ForwardPass.azsl",
+
+    "DepthStencilState" :
+    {
+        "Depth" :
+        {
+            "Enable" : true,
+            "CompareFunc" : "GreaterEqual"
+        },
+        "Stencil" :
+        {
+            "Enable" : true,
+            "ReadMask" : "0x00",
+            "WriteMask" : "0xFF",
+            "FrontFace" :
+            {
+                "Func" : "Always",
+                "DepthFailOp" : "Keep",
+                "FailOp" : "Keep",
+                "PassOp" : "Replace"
+            },
+            "BackFace" :
+            {
+                "Func" : "Always",
+                "DepthFailOp" : "Keep",
+                "FailOp" : "Keep",
+                "PassOp" : "Replace"
+            }
+        }
+    },
+
+    "ProgramSettings":
+    {
+      "EntryPoints":
+      [
+        {
+          "name": "VegetationBending_ForwardPassVS",
+          "type": "Vertex"
+        },
+        {
+          "name": "StandardPbr_ForwardPassPS_EDS",
```

Comment:
Can this shader run with EDS actually? I generally expect foliage to have transparency or alpha-to-coverage or other things going on. We at least need the non EDS option available also.

Reply:
The non-eds option is VegetationBending_ForwardPass.shader. We are following the same pattern as StandardPbr here. You can see when EDS is enabled vs. disabled in VegetationBending_ShaderEnable.lua, although taking a look at that here, we should be using StandardPBR_ShaderEnable.lua, instead of our own VegetationBending_ShaderEnable.lua since the logic is the same. Though that may not be possible to share the .lua functors across gems/projects at the moment.

AFAIK we don't have support for alpha-to-coverage in StandardPbr, so that might be something we should add. But this tutorial is about how to do custom vertex displacement while otherwise using StandardPbr shaders, so while we use vegetation as an example, the goal is not to replace the rest of the rendering logic, but to just replace the vertex shaders.